xapian-core/api/valuerangeproc.cc

```diff
@@ -439,4 +439,100 @@ NumberRangeProcessor::operator()(const string& b, const string& e)
     return Xapian::Query(Xapian::Query::OP_INVALID);
 }
 
+static const int size_map[26] = {
+    0,
+    1, // B
+    0, 0, 0, 0,
+    1024 * 1024 * 1024, // G
+    0, 0, 0,
+    1024, // K
+    0,
+    1024 * 1024, // M
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+};
+
+static int
+get_bytes(char ch)
+{
+	// It takes character as input and return number of bytes
+	// B - 1 byte
+	// K - 1024 bytes
+	// M - 1024*1024 bytes
+	// G - 1024*1024*1024 bytes
+	if (ch >= 'A' && ch <= 'Z' && size_map[ch - 'A'] > 0) {
+    return size_map[ch - 'A'];
```

Comment:
The alignment of the if and else legs is wrong here, but in two different ways. One of them suggests that your tab width is set to 4 when it should be 8, the other suggests you aren't trying to indent the legs further, as:

```c
if (…) {
    return …;
}
```